### PR TITLE
lava:tests: Add swupdate sanity check.

### PR DESCRIPTION
**Pull request by rwalton-arm**
Add a test which uses swupdate to install a dummy payload and ensures
swupdate calls a test shell script handler.

### CODE REVIEW

**Comment by rwalton-arm** [comment]
Successful run in LAVA: https://lava.mbedcloudtesting.com/scheduler/job/71724

File: ci/lava/tests/test-swupdate.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        )
+        subprocess.run(
+            [init_payload_cmd], check=True, shell=True, cwd=str(tmp_path)
+        )
+        append_payload_cmd = (
+            "echo {file_path} | cpio --append --format crc -o -F "
+            "{payload_path}"
+        )
+
+        subprocess.run(
```
Comment (petedyerarm) [inline ci/lava/tests/test-swupdate.py]: as above.

File: ci/lava/tests/test-swupdate.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                append_payload_cmd.format(
+                    file_path=str(test_file_path.name),
+                    payload_path=str(payload_path.name),
+                )
+            ],
+            check=True,
+            shell=True,
+            cwd=str(tmp_path),
+        )
+        subprocess.run(
```
Comment (petedyerarm) [inline ci/lava/tests/test-swupdate.py]: As above

File: ci/lava/tests/test-swupdate.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        payload_path = tmp_path / "payload.swu"
+
+        test_file_path.write_text("TEST")
+        sw_desc_path.write_text(SW_DESCRIPTION)
+        shellscript_path.write_text(SHELLSCRIPT)
+
+        init_payload_cmd = "echo {} | cpio --format crc -o -F {}".format(
+            str(sw_desc_path.name), str(payload_path)
+        )
+        subprocess.run(
```
Comment (petedyerarm) [inline ci/lava/tests/test-swupdate.py]: could this be done with execute_helper.execute_command ?

File: ci/lava/tests/test-swupdate.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    dut_addr = ""
+
+    def test_setup_dut_addr(self, dut_addr):
+        """Store the device address."""
+        TestSwupdate.dut_addr = dut_addr
+
+        assert dut_addr
+
+    def test_swupdate_calls_shellscript_handler(
+        self, execute_helper, tmp_path
```
[the remark this answers is not in the document]
Comment (rwalton-arm) [reply ci/lava/tests/test-swupdate.py]: `tmp_path` is a pytest default fixture http://doc.pytest.org/en/latest/tmpdir.html

File: ci/lava/tests/test-swupdate.py (hunk 1 above)
Comment (rwalton-arm) [reply ci/lava/tests/test-swupdate.py]: replied above

File: ci/lava/tests/test-swupdate.py (hunk 2 above)
Comment (rwalton-arm) [reply ci/lava/tests/test-swupdate.py]: replied above

File: ci/lava/tests/test-swupdate.py (hunk 3 above)
Comment (rwalton-arm) [reply ci/lava/tests/test-swupdate.py]: I need to set the `cwd` in some of these commands, which `execute_command` doesn't seem to support

Comment (petedyerarm) [reply ci/lava/tests/test-swupdate.py]: ok

**Pull request merged**